### ISSUE TITLE
基本没改代码，结果

### ISSUE THREAD

**Issue by xy123-l123** [yeyupiaoling/VoiceprintRecognition-Tensorflow#14]
![QMIQK7})G%FDFH18~JFDC3W](https://user-images.githubusercontent.com/75716456/112744415-8c9dfb80-8fd2-11eb-822c-fd6e793797c6.png)
这个结果是啥情况，我打印过test_loss 结果也是零点几，但是test_losses这个list打印出来几十项结果，前几项零点几，后面大部分就是5. 6. 9.多，是哪一部份有问题？ 要在哪修改？

**Comment by wenku1992abc** [comment]
你可以看一下 closed issue中的问题，你或许可以找到答案~

**Comment by xy123-l123** [comment]
重新训练这个模型的话，学习率多少比较合适？
学习率设定为0.001，在bath 1500左右的时候，训练集和测试集的损失和正确都不变了，结果不太正常。
之后我注释掉了l2和dropout层，而且把学习率改成了0.005，我要更改些什么才能
变成你上次发的图片那样，test集正确率百分之九十几？@wenku1992abc

**Comment by xy123-l123** [comment]
![P0}CYP_ R07Z(AX$C48YNUB](https://user-images.githubusercontent.com/75716456/112932679-e88e8e80-9150-11eb-88bd-57d28ddbb358.png)

**Comment by wenku1992abc** [comment]
感覺是你數據的問題，你可以看下dataset/文件夾中的數據 label正常不？

**Comment by xy123-l123** [comment]
@wenku1992abc  你好，
我看了生成的list.txt。感觉没出错。
我想问一下您在训练自己模型时，有做出什么比较重要的或者大型的更改代码吗？重新训练这个模型，test的accurary总是在0.5左右。

**Comment by yeyupiaoling** [comment]
@xy123-l123 看样子应该是模型过拟合了。你数据量是多少？

**Comment by xy123-l123** [comment]
纯新手，数据集用的是您推荐的哪个855人的11w音频，数据量.....




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年3月30日(星期二) 晚上6:31
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [yeyupiaoling/VoiceprintRecognition-Tensorflow] 基本没改代码，结果 (#14)





 
@xy123-l123 看样子应该是模型过拟合了。你数据量是多少？

**Comment by wenku1992abc** [comment]
> @wenku1992abc 你好，
> 我看了生成的list.txt。感觉没出错。
> 我想问一下您在训练自己模型时，有做出什么比较重要的或者大型的更改代码吗？重新训练这个模型，test的accurary总是在0.5左右。

我沒有做更改
‘’重新训练这个模型，test的accurary总是在0.5左右。‘’這個意思是上面講述的問題消失了是吧？
對於“test的accurary总是在0.5左右”，訓練時長不夠吧~用GPU了嗎

**Comment by wenku1992abc** [comment]
@xy123-l123 
![image](https://user-images.githubusercontent.com/33471245/113074010-ceaa8580-91fc-11eb-8b73-4d53fce9b4a1.png)
wod
我的大致是這樣~準確度最高也就92%左右~
好像也有過擬合~

**Comment by xy123-l123** [comment]
我将数据集复制出来一小部分，新建了文件夹，出来的结果如下，两位大佬帮我看看。
2个说话人的240条音频，     train loss=0.00000，accuracy=1.0    /// test loss=0 ,accuracy=1.0 
10个说话人的1200条音频， train loss=0.00000,  accuracy=1.0     /// test loss=0 ,accuracy=1.0      使用infer.py 对比音频，同一个 
     人，相似度0.8以上。不同的的人说话，相似度在0.7以上。相似度偏大，实际不是一个说话人也归纳与一个人了，。
36个说话人的4300条音频， train loss=0.000065，accuracy=1.0   ///test loss=0.13，accuracy=0.96    使用inter.py对比音频同一个 
    人说话，相似度在0.4到0.6。
48个说话人的5700条数据，目前正在跑 train loss=0.5到1.5，accuracy=0.8/0.9  ////test loss 4/5/3.*,accuracy=0.3  
但是当我更改optimizer = tf.keras.optimizers.Adam(learning_rate=0.001) 学习率时就会出现train loss=6/5， test loss=15.*  test accuracy =0.0001**,基本这样，不太好描述。

**Comment by xy123-l123** [comment]
最上面的问题没有消失，只是我搁置了，使用了init_model=None。
当使用博主的初始权重模型还是那样，读取这个权重，test loss=5/6, accuracy=0.5   ,accuracy无法更进一步
GPU耗尽，没有使用GPU。

**Comment by xy123-l123** [comment]
![L8I) $2O8HW8~M}JAG5( $1](https://user-images.githubusercontent.com/75716456/113095610-522c9c80-9226-11eb-94b3-37cdfe1c35d6.png)
这是什么情况？

**Comment by xy123-l123** [comment]
在训练过程中，test accuracy出现了大幅下滑，怎么回事，  accuray 0.2-0.3-0.3-0.4-0.5-**0.6-0.2**-0.3-0.5-0.4-0.5-**0.6-0.3**,这样了很多次，是学习率太大了吗

**Comment by LiuWeihao33** [comment]
请问一下，我在差不多第六个epoch开始训练集就已经正确率百分之百了，但是测试集的正确率还是很低，不过一直在上升，是过拟合了吗，这种情况是正常的吗？

**Comment by yeyupiaoling** [comment]
过拟合了，那肯定不正常的，数据量不够吧

**Comment by LiuWeihao33** [comment]
非常感谢！我用的数据就是您使用的那个855个分类的数据集，只不过把它分成了75%训练集和25%测试集，请问您当时训练时，是训练集和测试集的正确率都一起上升的吗？大概训练了多少个batch呢？




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年5月11日(星期二) 晚上8:28
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [yeyupiaoling/VoiceprintRecognition-Tensorflow] 基本没改代码，结果 (#14)





 
过拟合了，那肯定不正常的，数据量不够吧

**Comment by yeyupiaoling** [comment]
具体多少忘记了，

**Comment by hbo-lambda** [comment]
完全按照博主的实现操作了一遍，数据集也一样，但是实际的结果有点差。我只录了3段语音测试，男声女声的相似度0.4，男男相似度0.7，又试了另外两个人的语音，相似度达到了0.8，有点不科学

**Comment by yeyupiaoling** [comment]
@hbo-lambda 这个项目是我之前练习用的，多少有点不合理，但我已经升级了代码，在这个分支：https://github.com/yeyupiaoling/VoiceprintRecognition-Tensorflow/tree/develop

当还没有全部完成，要等训练结果。
